t/t2035-checkout-locked.sh

```diff
@@ -0,0 +1,44 @@
+#!/bin/sh
+
+test_description='checkout must be able to overwrite open files'
+. ./test-lib.sh
+
+test_expect_success 'setup' '
+
+	echo hello >world &&
+	mkdir dear &&
+	echo hello>dear/world &&
+	git add -A &&
+	git commit -m hi &&
+	git tag hi &&
+	git checkout -b bye &&
+	git mv dear cruel &&
+	echo goodbye >world &&
+	echo goodbye >cruel/world &&
+	git add -A &&
+	git commit -m bye
+'
+
+test_expect_success 'lock for read ' '
+
+	git checkout bye &&
+	exec 7<world &&
+	exec 8<cruel/world &&
+	git checkout hi &&
+	exec 7>&- &&
+	exec 8>&- &&
+	test "$(git diff-files --raw)" = ""
```

Comment:
Please use `test_cmp` instead. It prints a nice error message with the file diff if the test fails.

Reply:
In this case, even better:

```sh
git diff-files --raw >output &&
test_must_be_empty output
```

Reply:
Done